streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java

```diff
@@ -0,0 +1,518 @@
+            testMetricByName(listMetricStore, ALL_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, RANGE_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, RANGE_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, FLUSH_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, FLUSH_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, RESTORE_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, RESTORE_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, PUT_RATE, 2);
+            testMetricByName(listMetricStore, PUT_TOTAL, 2);
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_RATE, 2);
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_TOTAL, 2);
+            testMetricByName(listMetricStore, GET_RATE, 2);
+            testMetricByName(listMetricStore, DELETE_RATE, 2);
+            testMetricByName(listMetricStore, DELETE_TOTAL, 2);
+            testMetricByName(listMetricStore, PUT_ALL_RATE, 2);
+            testMetricByName(listMetricStore, PUT_ALL_TOTAL, 2);
+            testMetricByName(listMetricStore, ALL_RATE, 2);
+            testMetricByName(listMetricStore, ALL_TOTAL, 2);
+            testMetricByName(listMetricStore, RANGE_RATE, 2);
+            testMetricByName(listMetricStore, RANGE_TOTAL, 2);
+            testMetricByName(listMetricStore, FLUSH_RATE, 2);
+            testMetricByName(listMetricStore, FLUSH_TOTAL, 2);
+            testMetricByName(listMetricStore, RESTORE_RATE, 2);
+            testMetricByName(listMetricStore, RESTORE_TOTAL, 2);
+            return true;
+        } catch (final Throwable e) {
+            return false;
+        }
+    }
+
+    private boolean testCacheMetric() {
+        try {
+            final List<Metric> listMetricCache = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream().filter(m -> m.metricName().group().equals(STREAM_CACHE_NODE_METRICS)).collect(Collectors.toList());
+            testMetricByName(listMetricCache, HIT_RATIO_AVG, 6);
+            testMetricByName(listMetricCache, HIT_RATIO_MIN, 6);
+            testMetricByName(listMetricCache, HIT_RATIO_MAX, 6);
+            return true;
+        } catch (final Throwable e) {
+            return false;
+        }
+    }
+
+    private void testMetricByName(final List<Metric> listMetric, final String metricName, final int numMetric) {
+        try {
+            final List<Metric> metrics = listMetric.stream().filter(m -> m.metricName().name().equals(metricName)).collect(Collectors.toList());
+            Assert.assertEquals("Size of metrics of type:'" + metricName + "' must be equal to:" + numMetric + " but it's equal to " + metrics.size(), numMetric, metrics.size());
+            for (final Metric m : metrics) {
+                Assert.assertNotNull("Metric:'" + m.metricName() + "' must be not null", m.metricValue());
+            }
+        } catch (final Throwable e) {
```

Comment:
Do we need to capture-and-rethrow here? Since the caller will always capture and swallow, this seems unnecessary to me.